guides/source/association_basics.md

```diff
@@ -2724,3 +2724,156 @@ will run a query like:
+class Entry < ApplicationRecord
+  delegated_type :entryable, types: %w[ Message Comment ], dependent: :destroy
+end
+```
+
+The name of the `delegated_type` definition specifies `entryable`. Also, specified subclasses `Message`, `Comment` as the delegate.
+
+The `Entry` class has `entryable_type` and `entryable_id` fields. This is the field with the `_type`, `_id` suffixes added to the name `entryable` in the `delegated_type` definition.
+`entryable_type` stores the subclass name of the delegatee, and `entryable_id` stores the record id of the delegatee subclass.
+
+Next, define a module that implements `delegated types`. Declare the `entryable` specified in the `delegated_type` definition to `has_one` association to `entry`.
+
+```ruby
+module Entryable
+  extend ActiveSupport::Concern
+
+  included do
+    has_one :entry, as: :entryable, touch: true
+  end
+end
+```
+
+And then include the created module in your subclass.
+
+```ruby
+# Schema: messages[ id, subject, body, created_at, updated_at ]
+class Message < ApplicationRecord
+  include Entryable
+end
+
+# Schema: comments[ id, content, created_at, updated_at ]
+class Comment < ApplicationRecord
+  include Entryable
+end
+```
+
+This completes the definition of `delegated types`. `Entry` now implements the following methods:
+
+```ruby
+Entry#entryable_class # => Message or Comment
+Entry#entryable_name  # => "message" or "comment"
+Entry.messages        # => Entry.where(entryable_type: "Message")
+Entry#message?        # => true when entryable_type == "Message"
+Entry#message         # => returns the message record, when entryable_type == "Message", otherwise nil
+Entry#message_id      # => returns entryable_id, when entryable_type == "Message", otherwise nil
+Entry.comments        # => Entry.where(entryable_type: "Comment")
+Entry#comment?        # => true when entryable_type == "Comment"
+Entry#comment         # => returns the comment record, when entryable_type == "Comment", otherwise nil
+Entry#comment_id      # => returns entryable_id, when entryable_type == "Comment", otherwise nil
+```
+
+### Object creation
+
+When creating an `Entry` model, create a subclass object at the same time.
+This is created at the same time by specifying an object of subclass `Message` in the `entryable` of the `Entry` class.
+
+```ruby
+Entry.create! entryable: Message.new(subject: "hello!")
+```
+
+### Adding further delegation
+
+If you want to make further queries on `Entry`, you can define `delegates` and use polymorphism to delegate to subclasses.
+For example, to delegate the `title` method from `Entry` to its subclasses:
+
+```ruby
+# Schema: entries[ id, entryable_type, entryable_id, created_at, updated_at ]
+class Entry < ApplicationRecord
+  delegated_type :entryable, types: %w[ Message Comment ]
+  delegates :title, to: :entryable
+end
+
+# Schema: messages[ id, subject, body, created_at, updated_at ]
+class Message < ApplicationRecord
```

Comment:
```suggestion
class Message < ApplicationRecord
  include Entryable
```

We still need to include this right? It might be confusing if that line is suddenly missing.

Reply:
indeed. Added `include Entryable` to avoid confusion.

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2862